### PR TITLE
DOC: Added exported colors to colors.api

### LINKED ISSUE #25521: [Doc]: `TABLEAU_COLORS` `XKCD_COLORS` etc undocumented
### Documentation Link

https://matplotlib.org/devdocs/api/colors_api.html#module-matplotlib.colors

### Problem

The exported color arrays 
https://github.com/matplotlib/matplotlib/blob/31d83a869ee4edf5eeec80a0aea31ba46e7c6e79/lib/matplotlib/colors.py#L57

are not documented in: https://matplotlib.org/devdocs/api/colors_api.html#module-matplotlib.colors

I've always considered considered importing these directly to a stable part of the API, something supported by the usage in the examples: https://github.com/matplotlib/matplotlib/blob/31d83a869ee4edf5eeec80a0aea31ba46e7c6e79/galleries/examples/color/named_colors.py#L107

### Suggested improvement

_No response_

### PR DESCRIPTION
**Pull request by MikiPWata**
## PR Summary
closes #25521 
- documented exported colors array in colors.api

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

**Comment by tacaswell** [comment]
image


Looks good!

**Review by tacaswell — APPROVED** [review]

**Comment by tacaswell** [comment]
@MikiPWata Do you think it is worth adding a note that these are dictionaries mapping names -> hex values as strings?

**Comment by MikiPWata** [comment]
> @MikiPWata Do you think it is worth adding a note that these are dictionaries mapping names -> hex values as strings?

yes, I think it would be useful, and it might be even better if I add a reference to https://matplotlib.org/devdocs/gallery/color/named_colors.html
as well.... should I?

**Comment by tacaswell** [comment]
Yes please! You can keep pushing commits to this branch and it will automatically update.

File: doc/api/colors_api.rst [hunk 1]
```diff
@@ -71,3 +71,23 @@ Functions
    same_color
    get_named_colors_mapping
    make_norm_from_scale
+
+Exported colors
+---------------
+
+.. note::
```
Comment (tacaswell) [inline doc/api/colors_api.rst:78]: I don't think this needs to be in a note, can can just be normal prose.

Maybe something like

>  The data used to populate the :ref:`named colors` are exposed as dictionaries that map color name to hex string

?

Comment (MikiPWata) [reply doc/api/colors_api.rst]: Got it!

**Comment by MikiPWata** [comment]
I will squash all the commit once it's good to go

File: doc/api/colors_api.rst [hunk 2]
```diff
@@ -71,3 +71,17 @@ Functions
    same_color
    get_named_colors_mapping
    make_norm_from_scale
+
+Exported colors
+---------------
+
+The data used to populate the :doc:`/gallery/color/named_colors` are exposed
+as dictionaries that map color name to hex string.
```
Comment (QuLogic) [inline doc/api/colors_api.rst:79]:
```suggestion
as dictionaries that map color names to hex strings.
```

**Review by QuLogic — COMMENTED** [review]
LGTM otherwise; feel free to squash.

**Review by QuLogic — APPROVED** [review]

**Comment by QuLogic** [comment]
Thanks @MikiPWata! Congratulations on your first PR to Matplotlib :tada: We hope to hear from you again.

**Pull request merged**